### PR TITLE
xgboost-spark document

### PR DESCRIPTION
**Pull request by CodingCat**
@yanboliang

### CODE REVIEW

**Comment by CodingCat** [comment]
sorry for the delay, I got bad cold in these days, 

let's first finish it here and then upstream to dmlc repo

File: jvm-packages/xgboost4j-spark/docs/index.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+(based on maven build), step by step (structure of program)
+
+## Data Preparation
+
+Data Frame based
+
+## Training 
+
+talk about different objective functions for different purposes
```
Comment (yanboliang) [inline jvm-packages/xgboost4j-spark/docs/index.md]: Explain how to set params with native XGBoost way and Spark MLlib way.

File: jvm-packages/xgboost4j-spark/docs/index.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+### Current Version of Gang Scheduling
+
+based on spark even listener
```
Comment (yanboliang) [inline jvm-packages/xgboost4j-spark/docs/index.md]: What about explain ```numWorkers``` as well? I found this param is a little confused and relevant to scheduling.

**Review by yanboliang — COMMENTED** [review]
Looks very good overall, left two minor comments. Thanks.

File: jvm-packages/xgboost4j-spark/docs/index.md (hunk 2 above)
Comment (CodingCat) [reply jvm-packages/xgboost4j-spark/docs/index.md]: I think we can talk about it in training part? as Gang Scheduling is an non-default behavior while numWorkers is something you must care about when using the system

File: jvm-packages/xgboost4j-spark/docs/index.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+(3 - 4 sentences intro)
+
+# Build an Application with XGBoost4J-Spark
+
+(based on maven build), step by step (structure of program)
+
+## Data Preparation
+
+Data Frame based
```
Comment (CodingCat) [inline jvm-packages/xgboost4j-spark/docs/index.md]: @yanboliang I think we can choose a classic dataset here, maybe the ones here https://github.com/dmlc/xgboost/tree/master/demo/data

**Comment by CodingCat** [comment]
how you want to divide the workload? @yanboliang

File: jvm-packages/xgboost4j-spark/docs/index.md (hunk 3 above)
Comment (yanboliang) [reply jvm-packages/xgboost4j-spark/docs/index.md]: Sound good.

File: jvm-packages/xgboost4j-spark/docs/index.md (hunk 2 above)
Comment (yanboliang) [reply jvm-packages/xgboost4j-spark/docs/index.md]: Either is ok, if we can address them more clear.

**Comment by yanboliang** [comment]
I'll take the section after ```Prediction```(included), you're responsible for the remaining part. What dow you think about it?

**Comment by CodingCat** [comment]
sure,

**Comment by CodingCat** [comment]
@yanboliang FYI, I have uploaded the latest version to github-based artifact repo

you can refer the dependency as https://github.com/dmlc/xgboost/tree/master/jvm-packages#access-snapshot-version

**Pull request closed without merging**